extensions/ql-vscode/src/variant-analysis/view-autofixes.ts

```diff
@@ -0,0 +1,801 @@
+  storagePath: string,
+  app: App,
+  cliServer: CodeQLCliServer,
+): Promise<void> {
+  await withProgress(
+    async (progress: ProgressCallback) => {
+      // Get the path to the local autofix installation.
+      progress(progressUpdate(1, 4, `Checking for local autofix installation`));
+      const localAutofixPath = findLocalAutofix();
+
+      // Get the variant analysis with the given id.
+      const variantAnalysis = variantAnalyses.get(variantAnalysisId);
+      if (!variantAnalysis) {
+        throw new Error(`No variant analysis with id: ${variantAnalysisId}`);
+      }
+
+      // Generate the query help and output it to the override directory.
+      progress(progressUpdate(2, 4, `Generating query help override`));
+      await overrideQueryHelp(variantAnalysis, cliServer, localAutofixPath);
+
+      // Get the full names (nwos) of the selected repositories.
+      const selectedRepoNames = getSelectedRepositoryNames(
+        variantAnalysis,
+        filterSort,
+      );
+
+      // Get storage paths for the autofix results.
+      const {
+        variantAnalysisIdStoragePath,
+        sourceRootsStoragePath,
+        autofixOutputStoragePath,
+      } = await getStoragePaths(variantAnalysisId, storagePath);
+
+      // Process the selected repositories:
+      //  Get sarif
+      //  Download source root
+      //  Run autofix and output results
+      progress(
+        progressUpdate(
+          3,
+          4,
+          `Processing ${pluralize(selectedRepoNames.length, "repository", "repositories")}`,
+        ),
+      );
+      const outputTextFiles = await processSelectedRepositories(
+        selectedRepoNames,
+        variantAnalysisIdStoragePath,
+        sourceRootsStoragePath,
+        autofixOutputStoragePath,
+        localAutofixPath,
+        credentials,
+        logger,
+      );
+
+      // Output results from all repos to a combined markdown file.
+      progress(progressUpdate(4, 4, `Finalizing autofix results`));
+      const combinedOutputMarkdownFile = join(
+        autofixOutputStoragePath,
+        "autofix-output.md",
+      );
+      await mergeFiles(outputTextFiles, combinedOutputMarkdownFile, false);
+
+      // Open the combined markdown file.
+      await tryOpenExternalFile(app.commands, combinedOutputMarkdownFile);
+    },
+    {
+      title: "Generating Autofixes",
+      cancellable: false, // not cancellable for now
+    },
+  );
+}
+
+/**
+ * Finds the local autofix installation path from the AUTOFIX_PATH environment variable.
+ * Throws an error if the path is not set or does not exist.
+ * @returns An object containing the local autofix path.
+ * @throws Error if the AUTOFIX_PATH environment variable is not set or the path does not exist.
+ */
+function findLocalAutofix(): string {
+  const localAutofixPath = process.env.AUTOFIX_PATH;
+  if (!localAutofixPath) {
+    throw new Error("Path to local autofix installation not found.");
+  }
+  if (!pathExists(localAutofixPath)) {
```

Comment:
The `pathExists` function returns a promise, so this should be:


```suggestion
  if (!(await pathExists(localAutofixPath))) {
```